SecretManagement.LastPass.psm1

```diff
@@ -216,6 +216,85 @@ function Get-VaultParams {
 
 }
 
+<#
+.SYNOPSIS
+Show LastPass Grid view secrets then show the selected secret.
+
+.DESCRIPTION
+Show LastPass Grid view secrets then show the selected secret. 
+
+.PARAMETER Vault
+Name of the vault used for the lookup
+
+.PARAMETER Filter
+Pre-filter secrets based on the specified keywords
+
+.PARAMETER KeepOpen
+If set, the secrets GridView will be automatically reloaded after a secret is shown.
+
+.PARAMETER PassThru
+If set, Secret will be returned as is without formatting.
+
+.EXAMPLE
+Show-LastPassConsoleGridView -Vault MyVault -KeepOpen 
+
+.NOTES
+This cmdlet can make use of the improved Out-ConsoleGridView cmdlet if using Powershell 6.2 or newer and  Microsoft.PowerShell.ConsoleGuiTools is installed.
+Otherwise, Out-GridView will be used.
+#>
+Function Show-LastPassConsoleGridView {
+    [CmdletBinding(DefaultParameterSetName='Default')]
+    Param(
+        [String]$Vault,
+        [String]$Filter = '*',
+        [Parameter(ParameterSetName = 'KeepOpen')]
+        [Switch]$KeepOpen,
+        [Parameter(ParameterSetName = 'PassThru')]
+        [Switch]$PassThru
+    )
+    
+    $UseConsoleGridView = $false
+    try {
+        import-module 'Microsoft.PowerShell.ConsoleGuiTools' -ErrorAction Stop
+        $UseConsoleGridView = $true
+    }
+    catch {
+        Write-Debug "Microsoft.Powershell.ConsoleGuiTools could not be loaded.`n$($_ | Out-String)"
+    }
+
+    $Vault = (Get-SelectedVault -Vault $Vault).Name
+    $LastPassSecretInfoCache = Microsoft.Powershell.SecretManagement\Get-SecretInfo -Vault $Vault -Name "$Filter*"
+
+    do {
+        if ($UseConsoleGridView) {
+            $Result = $LastPassSecretInfoCache | Out-ConsoleGridView -Title "LastPass ($Vault)" -OutputMode Single
+        } else {
+            $Result = $LastPassSecretInfoCache | Out-GridView -Title "LastPass ($Vault)" -OutputMode Single
+        }
+        
+        if ($null -eq $Result) { break }
+        $Result | ForEach-Object { 
+            $Secret = Microsoft.Powershell.SecretManagement\Get-Secret -Vault $Vault -Name $_.Name -AsPlainText 
+            if ($PassThru) {return $Secret}
```

Comment:
do you need this if it gets emitted below?

Reply:
Not sure if still relevant with the last commit.
https://github.com/TylerLeonhardt/SecretManagement.LastPass/pull/23/commits/f695769b3ea13a3a53cadef962c1645b1bcf1158

The Formatted switch is there to provide a readable expanded output. 
Default output is now what the "Passthru" initially did.